src/common/tracer.h

```diff
@@ -0,0 +1,64 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+
+#include <jaegertracing/Tracer.h>
+
+//forward declaration of req_state defined /rgw/rgw_common.h
+struct req_state;
+
+typedef std::unique_ptr<opentracing::Span> Span;
+
+class Jager_Tracer{
+  public:
+    Jager_Tracer(){}
+    ~Jager_Tracer(){
+      if(this->tracer)
+        this->tracer->Close();
+    }
+
+    void init_tracer(const char* tracerName,const char* filePath){
+        auto yaml = YAML::LoadFile(filePath);
+        auto configuration = jaegertracing::Config::parse(yaml);
+        this->tracer = jaegertracing::Tracer::make(
+        tracerName,
```

Comment:
@suab321321 small thing, please amend to use snake_case eg: `tracer_name` when doing cleanup. I would encourage you to see check Ceph coding style and comply with it :)

Reply:
okay

Reply:
@suab321321 https://dev.to/yujiri8/snakecase-camelcase-or-dash-case-c0a